analyzers/src/SonarAnalyzer.CSharp/Common/Walkers/CatchLoggingInvocationWalker.cs

```diff
@@ -105,16 +99,6 @@ public override void VisitThrowStatement(ThrowStatementSyntax node)
         base.VisitThrowStatement(node);
     }
 
-    public override void VisitParenthesizedLambdaExpression(ParenthesizedLambdaExpressionSyntax node)
-    {
-        // Skip processing to avoid false positives.
-    }
-
-    public override void VisitSimpleLambdaExpression(SimpleLambdaExpressionSyntax node)
-    {
-        // Skip processing to avoid false positives.
-    }
-
     private bool RethrowsCaughtException(ThrowStatementSyntax node) =>
```

Comment:
From my understanding, the other rule does not care for throw statements.
I think you should override `VisitThrowStatement` only on the derived walker, for the log-and-rethrow rule exclusively.